src/pybind/mgr/orchestrator.py

```diff
@@ -303,6 +303,19 @@ def remove_osds(self, osd_ids):
         """
         raise NotImplementedError()
 
+    def blink_device_light(self, ident_fault, on, host, device_id):
+        # type: (str, bool, str, str) -> WriteCompletion
+        """
+        Instructs the orchestrator to enable or disable either the ident or the fault LED.
+
+        :param ident_fault: either ``"ident"`` or ``"fault"``
+        :param on: ``True`` = on.
+        :param host: hostname as in :func:`orchestrator.Orchestrator.get_hosts`
+        :param device_id: e.g. ``ABC1234DEF567-1R1234_ABC8DE0Q``.
```

Comment:
I was assuming we'd use the mapping we have in the device table so that the devname (e.g., ``sdb``) would be passed here.

Reply:
will do

Reply:
well, at least the deepsea orchestrator also can handle the device ids. I'd pass what ever is more convenient for the orchestrator module.

Reply:
The deepsea orchestrator handles device IDs because it was based on the blinky-lights pad which had device IDs as command line parameters, and this piece in the orchestrator wasn't implemented yet :-)

I'd be in favour of passing devnames here, on the assumption that that's the simplest thing for any orchestrator to implement, as it just becomes a call to `lsmcli [...] --path /dev/${devname}` internally (to handle device IDs, each individual orchestrator would first have to go digging around in `/dev/disk/by-id` looking for something that matches the ID, to figure out the actual path).

Reply:
> Also, the current implementation doesn't really work, if we also want to allow to enable lights for devices not yet known to ceph

Will users generally expect to be able to invoke a `ceph [...]` command to manipulate devices ceph doesn't yet know about?

Reply:
> Will users generally expect to be able to invoke a ceph [...] command to manipulate devices ceph doesn't yet know about?

What about new and broken devices that I'd like to replace? I this case, there is no OSD running on that device.

Reply:
Currently ceph only learns teh device -> host+devname mapping when an osd or mon uses the device.  It doesn't forget about that mapping until teh same mon/osd comes up on another device, or the osd is purged from the system.  So it'll work for down osds.

In the future, we could make ceph discover the same mapping from other sources (like the orchestrator inventory) so that we can blink lights that osds have never touched.  Not trying to solve that problem now, though.

Reply:
In the realm of the orchestrator, it turned out to be often a bit easier, if we don't need to look that much into Ceph's data structures. By just relying on the external orchestrators, we don't need to keep anything synchronized with Ceph. 

From a Dashboard user's point of view (if we're showing the inventory and provide a way to enable device lights), it would make sense to be able to trigger the lights for all devices displayed in the Dashboard, instead of only allowing the ones known to Ceph. This was my main use case for allowing all devices right now.

Sorry for my ignorance here!It just feels odd to me to limit ourselves to devices know to Ceph at this point. 

If the mapping of dev_id to dev_name + host turns out to be problematic, an alternative would then be to just skip this mapping altogether and just rely on name+host in the module's mapping. 

Another alternative would be to query the orchestrator's inventory to determine the device_id for unknown ids and only store the device_id in the module. This should actually not complicate things that much.

Reply:
Yeah, I think if the orchestrator implemenation takes a devid, that makes the per-orch plumbing trivial.  That's the argument for devname at that layer.

The open question is then how the 'ceph device ident-light-on ...' commands work when teh device isn't known yet.  Notably we have what we need from the orchestrator inventory, so perhaps the evolution of this is to store the 'on' sets in terms of ``host:devname`` instead of devid and then add a CLI variant that can take directly (instead of a devid).  Or, perhaps better, have the orchestrator inventory populate the devices table in the mgr to include unbound devices (this would be my preference).

At this stage, though, my preference is to (1) keep the per-orch implementations simple (take devname) and (2) start with what we already have with known devices, and extend this to unbound devices later.

Reply:
updated.